pr_agent/algo/utils.py

```diff
@@ -211,3 +211,27 @@ def load_large_diff(file, new_file_content_str: str, original_file_content_str:
         except Exception:
             pass
     return patch
+
+
+def update_settings_from_args(args):
+    if args and len(args) >= 1:
+        for arg in args:
+            try:
+                arg = arg.strip('-').strip()
+                vals = arg.replace('=', '.').replace('=', '.').split('.')
+                d = settings
+                for i, v in enumerate(vals[:-1]):
+                    if i == len(vals) - 2:
+                        if v in d:
+                            if type(d[v]) == bool:
+                                d[v] = vals[-1].lower() in ("yes", "true", "t", "1")
+                            else:
+                                d[v] = type(d[v])(vals[-1])
+                            logging.info(f'Updated setting {vals[:-1]} to: "{vals[-1]}"')
+                            break
+                        else:
+                            logging.error(f'Invalid setting {vals[:-1]}')
+                    else:
+                        d = d[v]
+            except Exception as e:
+                logging.error(f'Failed to parse argument {arg}: {e}')
```

Comment:
**Suggestion:** The `update_settings_from_args` function does not handle the case where the value of the setting is not a boolean or a type that can be directly cast from a string. Consider adding a check for this case to avoid runtime errors.
```suggestion
def update_settings_from_args(args):
    if args and len(args) >= 1:
        for arg in args:
            try:
                arg = arg.strip('-').strip()
                vals = arg.replace('=', '.').replace('=', '.').split('.')
                d = settings
                for i, v in enumerate(vals[:-1]):
                    if i == len(vals) - 2:
                        if v in d:
                            if type(d[v]) == bool:
                                d[v] = vals[-1].lower() in ("yes", "true", "t", "1")
                            elif isinstance(d[v], (int, float, str)):
                                d[v] = type(d[v])(vals[-1])
                            else:
                                logging.error(f'Cannot cast {vals[-1]} to type {type(d[v])}')
                                break
                            logging.info(f'Updated setting {vals[:-1]} to: "{vals[-1]}"')
                            break
                        else:
                            logging.error(f'Invalid setting {vals[:-1]}')
                    else:
                        d = d[v]
            except Exception as e:
                logging.error(f'Failed to parse argument {arg}: {e}')
```